user_sync/post_sync/connectors/sign_sync/client.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import requests
+import json
+
+
+class SignClient:
+    version = 'v5'
+    _endpoint_template = 'api/rest/{}/'
+    DEFAULT_GROUP_NAME = 'default group'
+
+    def __init__(self, config):
+        for k in ['host', 'key', 'admin_email']:
+            assert k in config, "Key '{}' must be specified for all Sign orgs".format(k)
+        self.host = config['host']
+        self.key = config['key']
+        self.admin_email = config['admin_email']
+        self.console_org = config['console_org'] if 'console_org' in config else None
+        self.api_url = self.base_uri()
+        self.groups = self.get_groups()
```

Comment:
The question of whether to call the sign API to query groups prior to running the sync tool.  Does it make more sense to call this method in the run() call so as to avoid 

(1) not breaking the normal sync if the call fails for some reason
(2) a potentially very long delay between when groups are fetched from sign and when they are actually used

I think its good to keep the base_uri() call since it provides immediate feedback if the api key is wrong.

OR - in contrast to the above, does it make sense to add some kind of post sync validation code to check the configuration of a post sync so we know the authentication will be good when the time comes.  Again, I'm just thinking in terms of large scale / long sync processes.  Might be overkill.